src/main/java/supercoder79/simplexterrain/world/gen/SimplexBiomeSource.java

```diff
@@ -41,40 +46,17 @@ public void setHeightmap(Heightmap heightmap) {
 
 	@Override
 	public Biome getBiomeForNoiseGen(int x, int j, int z) {
-		return sampleBiomeWithMathTM((x << 2), (z << 2), heightmap.getHeight((x << 2), (z << 2)));
+		return sampleBiomeWithMathTM(x, z, heightmap.getHeight((x << 2), (z << 2)));
 	}
 
 	public Biome sampleBiomeWithMathTM(int x, int z, int height) {
-		if (height < SimplexTerrain.CONFIG.seaLevel/2) {
-			double oceanTemperatureSample = oceanTemperatureLayer.sample(x, z);
-			if (oceanTemperatureSample > 0.6)
-				return Biomes.DEEP_WARM_OCEAN;
-			else if (oceanTemperatureSample > 0.3)
-				return Biomes.DEEP_LUKEWARM_OCEAN;
-			else if (oceanTemperatureSample < -0.3)
-				return Biomes.DEEP_FROZEN_OCEAN;
-			else if (oceanTemperatureSample < -0.6)
-				return Biomes.DEEP_COLD_OCEAN;
-
-			else return Biomes.DEEP_OCEAN;
-		}
-		if (height < SimplexTerrain.CONFIG.seaLevel-2) {
-			double oceanTemperatureSample = oceanTemperatureLayer.sample(x, z);
-			if (oceanTemperatureSample > 0.6)
-				return Biomes.WARM_OCEAN;
-			else if (oceanTemperatureSample > 0.3)
-				return Biomes.LUKEWARM_OCEAN;
-			else if (oceanTemperatureSample < -0.6)
-				return Biomes.FROZEN_OCEAN;
-			else if (oceanTemperatureSample < -0.3)
-				return Biomes.COLD_OCEAN;
-
-			else return Biomes.OCEAN;
-		}
-		if (height < SimplexTerrain.CONFIG.lowlandStartHeight) return Biomes.BEACH;
-		if (height < SimplexTerrain.CONFIG.midlandStartHeight)  return lowlandsSampler.sample(x, z);
-		if (height < SimplexTerrain.CONFIG.highlandStartHeight) return midlandsSampler.sample(x, z);
-		if (height < SimplexTerrain.CONFIG.toplandStartHeight) return highlandsSampler.sample(x, z);
-		return toplandsSampler.sample(x, z);
+		if (height < SimplexTerrain.CONFIG.seaLevel / 2) return this.deepOceanSampler.sample(x, z);
+		if (height < SimplexTerrain.CONFIG.seaLevel - 7) return this.oceanSampler.sample(x, z);
+		if (height < SimplexTerrain.CONFIG.seaLevel - 2) return Biomes.OCEAN;
```

Comment:
why does it return ocean here

Reply:
so that water colour on biome edges don't look weird

Reply:
suddenly getting coloured a few blocks out

Reply:
you can tweak the cutoff values /shrug

Reply:
the beaches already do that right